core/src/components/tab-button/tab-button.ionic.scss

```
@@ -86,3 +88,52 @@
 :host(.tab-button-shape-rectangular) {
   @include globals.border-radius(globals.$ion-border-radius-0);
 }
+
+// Tab Button: Badge
+// --------------------------------------------------
+
+:host ::slotted(ion-badge) {
+  @include globals.position(null, null, null, calc(50% + globals.$ion-scale-300));
+}
+
+:host ::slotted(ion-badge.badge-vertical-top) {
+  @include globals.position(-2px, null, null, null);
+}
+
+:host ::slotted(ion-badge.badge-vertical-top:empty) {
+  @include globals.position(calc(globals.$ion-scale-100 * -1), null, null, null);
+}
+
+:host ::slotted(ion-badge.badge-vertical-bottom) {
+  @include globals.position(calc(50% - globals.$ion-scale-200), null, null, null);
+}
+
+:host ::slotted(ion-badge.badge-vertical-bottom:empty) {
+  @include globals.position(calc(50% - globals.$ion-scale-100), null, null, null);
```

Since we don't change any of the other value, we can depend on the defaults. I recommend reviewing how `position` works.

```suggestion
  @include globals.position(calc(50% - globals.$ion-scale-100));
```

---

Same as before!